src/coop/rchain/lang/Rho.bnf

```diff
@@ -106,23 +106,28 @@ private stringLit ::= STRING_LIT | RAW_STRING_LIT | CHAR_LIT
 private comment ::= (BLOCK_COMMENT|BLOCK_DOC_COMMENT|LINE_COMMENT|LINE_DOC_COMMENT)
 
 // Processes
-Proc4 ::= NIL | Value
-Proc3 ::= ChanRefSymbol Chan ["(" [Proc] ")"]
-Proc2 ::= Chan "!" "(" [Proc] ")"
-Proc1 ::= SUM "(" Bind "/:" Bind ")" ProcBlock
+private Proc4 ::= NIL | Value
+private Proc3 ::= ChanRefSymbol Chan [ParenthesizedProc]
+private Proc2 ::= Chan "!" ParenthesizedProc
+private Proc1 ::= SUM "(" Bind "/:" Bind ")" ProcBlock
         | TOTAL "(" Bind ":\\" Bind ")" ProcBlock
-        | FOR "(" [Bind] ")" "{" [Proc] "}"
+        | FOR ForParamsList ProcBlock
         | SELECT "{" [CBranch] "}"
-        | MATCH Proc WITH [PMBranch]
+        | MatchWith
         | MATCH Proc { [PMBranch] }
         | NEW [VarPattern] IN ProcBlock
         | NEW [VarPattern] IN Proc
         | LET ("(" CPattern ")" | CPattern) "=" Proc IN Proc
-        | ConstrName "(" [Proc] ")"
-        | ID_NAME "(" [Proc] ")"
+        | ConstrName ParenthesizedProc
+        | ID_NAME ParenthesizedProc
         | Contract_
         | "@" ID_NAME
-Proc_ ::= (FnDef | Proc1 | Proc2 | Proc3 | Proc4 | ConstrName) [TypePattern] | comment
+
+ParenthesizedProc::= "(" [Proc] ")"
+MatchWith ::= MATCH Proc WITH [PMBranch]
+
+ForParamsList ::= "(" [Bind] ")"
```

Comment:
You already replaced `"{" [Proc] "}"` with `ProcBlock`. It looks like a similar situation now but with parenthesis and very possible for other cases in the future. I'm thinking that we can have a naming rule like:
```
ProcBlock  ::= "{" [Proc] "}"
ProcParen ::= "(" [Proc] ")"
BindParen ::= "(" [Bind] ")"
```
And also `CBranchBlock ::= "{" [CBranch] "}"` to support formatting of select-case block.

Reply:
Aha, actually I have these BindParen and ProcParen with different name, as ForParamsList ::= "(" [Bind] ")" and  ParenthesizedProc::= "(" [Proc] ")".  Your suggestion is more consistent.